File: src/main/java/ma/fstg/gestionprojets/services/AffectationService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.request.AffectationRequest;
import ma.fstg.gestionprojets.dto.response.AffectationResponse;
import java.util.List;
public interface AffectationService {
    AffectationResponse create(Long phaseId, Long employeId, AffectationRequest req);
    AffectationResponse update(Long phaseId, Long employeId, AffectationRequest req);
    AffectationResponse getOne(Long phaseId, Long employeId);
    List<AffectationResponse> getByPhase(Long phaseId);
    List<AffectationResponse> getByEmploye(Long employeId);
    List<AffectationResponse> getAll();
    void delete(Long phaseId, Long employeId);
}

File: gestion-projets-frontend/src/services/affectationService.ts
import api from '../utils/axiosConfig';

export const affectationService = {
  getByPhase: (phaseId: number) => api.get(`/phases/${phaseId}/employes`),
  getByEmploye: (employeId: number) => api.get(`/employes/${employeId}/phases`),
  getOne: (phaseId: number, employeId: number) => api.get(`/phases/${phaseId}/employes/${employeId}`),
  create: (phaseId: number, employeId: number, data: any) =>
    api.post(`/phases/${phaseId}/employes/${employeId}`, data),
  update: (phaseId: number, employeId: number, data: any) =>
    api.put(`/phases/${phaseId}/employes/${employeId}`, data),
  delete: (phaseId: number, employeId: number) =>
    api.delete(`/phases/${phaseId}/employes/${employeId}`),
};
